lib/elasticsearch/indexing/bulk.ex

```diff
@@ -51,9 +51,9 @@ defmodule Elasticsearch.Index.Bulk do
       iex> Bulk.encode!(Cluster, 123, "my-index")
       ** (Protocol.UndefinedError) protocol Elasticsearch.Document not implemented for 123. This protocol is implemented for: Comment, Post
   """
-  def encode!(cluster, struct, index) do
+  def encode!(cluster, struct, index, action \\ "create") do
```

Comment:
It seems there should be an additional function head here for `"delete"`, which provides the header only.

> `delete` does not expect a source on the next line and has the same semantics as the standard delete API.
https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html

Reply:
@erikreedstrom seems reasonable to me. Our current use cases don't include bulk deletes so we're currently using the document delete API.